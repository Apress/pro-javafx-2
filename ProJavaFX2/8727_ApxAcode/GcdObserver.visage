/*
 * @test
 * @run
 */

class Gcd {
  var a: Integer;
  var b: Integer;
  function gcd(): Integer {
    if (b == 0) {
      return a;
    } else {
      var tmp = a mod b;
      a = b;
      b = tmp;
      return gcd();
    }
  }
}

var o = Gcd { a: 165, b: 105 };

var x = bind o.b on replace {
  println("Iterating: o.a={o.a}, o.b = {x}.");
}

var result = o.gcd();
println("The gcd of 106 and 105 is {result}.");
