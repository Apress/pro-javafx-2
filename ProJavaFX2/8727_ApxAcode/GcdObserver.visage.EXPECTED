Iterating: o.a=165, o.b = 105.
Iterating: o.a=105, o.b = 60.
Iterating: o.a=60, o.b = 45.
Iterating: o.a=45, o.b = 15.
Iterating: o.a=15, o.b = 0.
The gcd of 106 and 105 is 15.
